src/Init/Data/Vector/Monadic.lean

```diff
@@ -29,6 +29,12 @@ open Nat
 
 /-! ### mapM -/
 
+@[simp]
+theorem mapM_pure [Monad m] [LawfulMonad m] {xs : Vector α n} (f : α → β) :
```

Comment:
@kim-em, do you want this in `Monadic` (which claims to only be about `forIn`), or in `Lemmas` which is where many other monadic lemmas are?

Reply:
Leave it in `Monadic` for now. I'll need to rehome these lemmas sometime.

Reply:
Thanks for taking on the rehoming!